[debug] [2024-09-15T03:18:55.417Z] ----------------------------------------------------------------------
[debug] [2024-09-15T03:18:55.419Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\wou12\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only hosting
[debug] [2024-09-15T03:18:55.419Z] CLI Version:   13.17.0
[debug] [2024-09-15T03:18:55.419Z] Platform:      win32
[debug] [2024-09-15T03:18:55.419Z] Node Version:  v20.12.1
[debug] [2024-09-15T03:18:55.419Z] Time:          Sat Sep 14 2024 20:18:55 GMT-0700 (Pacific Daylight Time)
[debug] [2024-09-15T03:18:55.419Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-09-15T03:18:55.460Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-09-15T03:18:55.460Z] > authorizing via signed-in user (ungwilson94@gmail.com)
[debug] [2024-09-15T03:18:55.460Z] [iam] checking project ecommerce-store-1c980 for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-09-15T03:18:55.462Z] >>> [apiv2][query] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions [none]
[debug] [2024-09-15T03:18:55.462Z] >>> [apiv2][(partial)header] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions x-goog-quota-user=projects/ecommerce-store-1c980
[debug] [2024-09-15T03:18:55.462Z] >>> [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:18:55.694Z] <<< [apiv2][status] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions 200
[debug] [2024-09-15T03:18:55.694Z] <<< [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:18:55.695Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:18:55.840Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 200
[debug] [2024-09-15T03:18:55.841Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 {"projectId":"ecommerce-store-1c980","projectNumber":"373020330151","displayName":"eCommerce Store","name":"projects/ecommerce-store-1c980","resources":{"hostingSite":"ecommerce-store-1c980","realtimeDatabaseInstance":"ecommerce-store-1c980-default-rtdb"},"state":"ACTIVE","etag":"1_6d11945f-ce21-4170-8bf3-54b3d6926fdc"}
[debug] [2024-09-15T03:18:58.481Z] defaultcredentials: writing to file C:\Users\wou12\AppData\Roaming\firebase\ungwilson94_gmail_com_application_default_credentials.json
[debug] [2024-09-15T03:18:58.482Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites pageToken=&pageSize=10
[debug] [2024-09-15T03:18:59.182Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites 200
[debug] [2024-09-15T03:18:59.182Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites {"sites":[{"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}]}
[debug] [2024-09-15T03:18:59.182Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config [none]
[debug] [2024-09-15T03:18:59.584Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config 200
[debug] [2024-09-15T03:18:59.584Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config {"projectId":"ecommerce-store-1c980","appId":"1:373020330151:web:f8722f46f71b5b4664b382","databaseURL":"https://ecommerce-store-1c980-default-rtdb.firebaseio.com","storageBucket":"ecommerce-store-1c980.appspot.com","apiKey":"AIzaSyBMCAO2C6-rL8W8uQm5rIiSK3fIPPgdkNQ","authDomain":"ecommerce-store-1c980.firebaseapp.com","messagingSenderId":"373020330151"}
[info] 
   Thank you for trying our early preview of Next.js support on Firebase Hosting.
   During the preview, support is best-effort and breaking changes can be expected. Proceed with caution.

   Documentation: https://firebase.google.com/docs/hosting/frameworks/nextjs
   File a bug: https://github.com/firebase/firebase-tools/issues/new?template=bug_report.md
   Submit a feature request: https://github.com/firebase/firebase-tools/issues/new?template=feature_request.md

   We'd love to learn from you. Express your interest in helping us shape the future of Firebase Hosting: https://goo.gle/41enW5X

[debug] [2024-09-15T03:19:04.735Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:19:05.267Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 200
[debug] [2024-09-15T03:19:05.267Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 {"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}
[info] Error configuring redirects: TypeError: fetch failed

[info]    Creating an optimized production build...

[info]  ✓ Compiled successfully

[info]    Skipping validation of types

[info]    Linting...

[info] 
./src/app/_components/Promotion/index.tsx
37:6  Warning: React Hook useEffect has a missing dependency: 'targetDate'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules

[info]    Collecting page data...

[info]    Generating static pages (0/26)...

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]  ⚠ Entire page /reset-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /reset-password

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\8723.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info]    Generating static pages (6/26)


[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]    Generating static pages (12/26)


[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info]    Generating static pages (19/26)


[info] 
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\8723.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info] 
Error occurred prerendering page "/products". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info]  ✓ Generating static pages (26/26)

[info] 

[info] > Export encountered errors on following paths:
	/(pages)/products/page: /products
	/page: /

[debug] [2024-09-15T03:19:12.173Z] Error: ENOENT: no such file or directory, open 'C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\export-marker.json'
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2024-09-15T03:21:14.926Z] ----------------------------------------------------------------------
[debug] [2024-09-15T03:21:14.928Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\wou12\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy --only hosting
[debug] [2024-09-15T03:21:14.928Z] CLI Version:   13.17.0
[debug] [2024-09-15T03:21:14.928Z] Platform:      win32
[debug] [2024-09-15T03:21:14.928Z] Node Version:  v20.12.1
[debug] [2024-09-15T03:21:14.928Z] Time:          Sat Sep 14 2024 20:21:14 GMT-0700 (Pacific Daylight Time)
[debug] [2024-09-15T03:21:14.928Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-09-15T03:21:14.967Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-09-15T03:21:14.968Z] > authorizing via signed-in user (ungwilson94@gmail.com)
[debug] [2024-09-15T03:21:14.968Z] [iam] checking project ecommerce-store-1c980 for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-09-15T03:21:14.969Z] >>> [apiv2][query] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions [none]
[debug] [2024-09-15T03:21:14.969Z] >>> [apiv2][(partial)header] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions x-goog-quota-user=projects/ecommerce-store-1c980
[debug] [2024-09-15T03:21:14.969Z] >>> [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:21:15.216Z] <<< [apiv2][status] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions 200
[debug] [2024-09-15T03:21:15.216Z] <<< [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:21:15.216Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:21:15.395Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 200
[debug] [2024-09-15T03:21:15.395Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 {"projectId":"ecommerce-store-1c980","projectNumber":"373020330151","displayName":"eCommerce Store","name":"projects/ecommerce-store-1c980","resources":{"hostingSite":"ecommerce-store-1c980","realtimeDatabaseInstance":"ecommerce-store-1c980-default-rtdb"},"state":"ACTIVE","etag":"1_6d11945f-ce21-4170-8bf3-54b3d6926fdc"}
[debug] [2024-09-15T03:21:17.990Z] defaultcredentials: writing to file C:\Users\wou12\AppData\Roaming\firebase\ungwilson94_gmail_com_application_default_credentials.json
[debug] [2024-09-15T03:21:17.992Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites pageToken=&pageSize=10
[debug] [2024-09-15T03:21:18.693Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites 200
[debug] [2024-09-15T03:21:18.693Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites {"sites":[{"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}]}
[debug] [2024-09-15T03:21:18.694Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config [none]
[debug] [2024-09-15T03:21:19.200Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config 200
[debug] [2024-09-15T03:21:19.200Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config {"projectId":"ecommerce-store-1c980","appId":"1:373020330151:web:f8722f46f71b5b4664b382","databaseURL":"https://ecommerce-store-1c980-default-rtdb.firebaseio.com","storageBucket":"ecommerce-store-1c980.appspot.com","apiKey":"AIzaSyBMCAO2C6-rL8W8uQm5rIiSK3fIPPgdkNQ","authDomain":"ecommerce-store-1c980.firebaseapp.com","messagingSenderId":"373020330151"}
[info] 
   Thank you for trying our early preview of Next.js support on Firebase Hosting.
   During the preview, support is best-effort and breaking changes can be expected. Proceed with caution.

   Documentation: https://firebase.google.com/docs/hosting/frameworks/nextjs
   File a bug: https://github.com/firebase/firebase-tools/issues/new?template=bug_report.md
   Submit a feature request: https://github.com/firebase/firebase-tools/issues/new?template=feature_request.md

   We'd love to learn from you. Express your interest in helping us shape the future of Firebase Hosting: https://goo.gle/41enW5X

[debug] [2024-09-15T03:21:24.375Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:21:24.929Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 200
[debug] [2024-09-15T03:21:24.930Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 {"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}
[info] Error configuring redirects: TypeError: fetch failed

[info]    Creating an optimized production build...

[info]  ✓ Compiled successfully

[info]    Skipping validation of types
   Linting...

[info] 
./src/app/_components/Promotion/index.tsx
37:6  Warning: React Hook useEffect has a missing dependency: 'targetDate'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules

[info]    Collecting page data...

[info]    Generating static pages (0/26)...

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]  ⚠ Entire page /reset-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /reset-password

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\9861.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]    Generating static pages (6/26)


[info]    Generating static pages (12/26)


[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] 
Error occurred prerendering page "/products". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info]    Generating static pages (19/26)


[info] 
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\9861.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info]  ✓ Generating static pages (26/26)

[info] 

[info] > Export encountered errors on following paths:
	/(pages)/products/page: /products
	/page: /

[debug] [2024-09-15T03:21:31.536Z] Error: ENOENT: no such file or directory, open 'C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\export-marker.json'
[error] 
[error] Error: An unexpected error has occurred.
[debug] [2024-09-15T03:21:49.277Z] ----------------------------------------------------------------------
[debug] [2024-09-15T03:21:49.278Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\wou12\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy
[debug] [2024-09-15T03:21:49.279Z] CLI Version:   13.17.0
[debug] [2024-09-15T03:21:49.279Z] Platform:      win32
[debug] [2024-09-15T03:21:49.279Z] Node Version:  v20.12.1
[debug] [2024-09-15T03:21:49.279Z] Time:          Sat Sep 14 2024 20:21:49 GMT-0700 (Pacific Daylight Time)
[debug] [2024-09-15T03:21:49.279Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-09-15T03:21:49.318Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-09-15T03:21:49.318Z] > authorizing via signed-in user (ungwilson94@gmail.com)
[debug] [2024-09-15T03:21:49.318Z] [iam] checking project ecommerce-store-1c980 for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-09-15T03:21:49.319Z] >>> [apiv2][query] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions [none]
[debug] [2024-09-15T03:21:49.319Z] >>> [apiv2][(partial)header] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions x-goog-quota-user=projects/ecommerce-store-1c980
[debug] [2024-09-15T03:21:49.319Z] >>> [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:21:49.556Z] <<< [apiv2][status] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions 200
[debug] [2024-09-15T03:21:49.556Z] <<< [apiv2][body] POST https://cloudresourcemanager.googleapis.com/v1/projects/ecommerce-store-1c980:testIamPermissions {"permissions":["firebase.projects.get","firebasehosting.sites.update"]}
[debug] [2024-09-15T03:21:49.557Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:21:49.730Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 200
[debug] [2024-09-15T03:21:49.730Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/ecommerce-store-1c980 {"projectId":"ecommerce-store-1c980","projectNumber":"373020330151","displayName":"eCommerce Store","name":"projects/ecommerce-store-1c980","resources":{"hostingSite":"ecommerce-store-1c980","realtimeDatabaseInstance":"ecommerce-store-1c980-default-rtdb"},"state":"ACTIVE","etag":"1_6d11945f-ce21-4170-8bf3-54b3d6926fdc"}
[debug] [2024-09-15T03:21:52.331Z] defaultcredentials: writing to file C:\Users\wou12\AppData\Roaming\firebase\ungwilson94_gmail_com_application_default_credentials.json
[debug] [2024-09-15T03:21:52.332Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites pageToken=&pageSize=10
[debug] [2024-09-15T03:21:52.912Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites 200
[debug] [2024-09-15T03:21:52.912Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites {"sites":[{"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}]}
[debug] [2024-09-15T03:21:52.913Z] >>> [apiv2][query] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config [none]
[debug] [2024-09-15T03:21:53.719Z] <<< [apiv2][status] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config 200
[debug] [2024-09-15T03:21:53.719Z] <<< [apiv2][body] GET https://firebase.googleapis.com/v1beta1/projects/-/webApps/1:373020330151:web:f8722f46f71b5b4664b382/config {"projectId":"ecommerce-store-1c980","appId":"1:373020330151:web:f8722f46f71b5b4664b382","databaseURL":"https://ecommerce-store-1c980-default-rtdb.firebaseio.com","storageBucket":"ecommerce-store-1c980.appspot.com","apiKey":"AIzaSyBMCAO2C6-rL8W8uQm5rIiSK3fIPPgdkNQ","authDomain":"ecommerce-store-1c980.firebaseapp.com","messagingSenderId":"373020330151"}
[info] 
   Thank you for trying our early preview of Next.js support on Firebase Hosting.
   During the preview, support is best-effort and breaking changes can be expected. Proceed with caution.

   Documentation: https://firebase.google.com/docs/hosting/frameworks/nextjs
   File a bug: https://github.com/firebase/firebase-tools/issues/new?template=bug_report.md
   Submit a feature request: https://github.com/firebase/firebase-tools/issues/new?template=feature_request.md

   We'd love to learn from you. Express your interest in helping us shape the future of Firebase Hosting: https://goo.gle/41enW5X

[debug] [2024-09-15T03:21:58.741Z] >>> [apiv2][query] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 [none]
[debug] [2024-09-15T03:21:59.275Z] <<< [apiv2][status] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 200
[debug] [2024-09-15T03:21:59.276Z] <<< [apiv2][body] GET https://firebasehosting.googleapis.com/v1beta1/projects/ecommerce-store-1c980/sites/ecommerce-store-1c980 {"name":"projects/ecommerce-store-1c980/sites/ecommerce-store-1c980","defaultUrl":"https://ecommerce-store-1c980.web.app","appId":"1:373020330151:web:f8722f46f71b5b4664b382","type":"DEFAULT_SITE"}
[info] Error configuring redirects: TypeError: fetch failed

[info]    Creating an optimized production build...

[info]  ✓ Compiled successfully

[info]    Skipping validation of types
   Linting...

[info] 
./src/app/_components/Promotion/index.tsx
37:6  Warning: React Hook useEffect has a missing dependency: 'targetDate'. Either include it or remove the dependency array.  react-hooks/exhaustive-deps

info  - Need to disable some ESLint rules? Learn more here: https://nextjs.org/docs/basic-features/eslint#disabling-rules

[info]    Collecting page data...

[info]    Generating static pages (0/26)...

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]  ⚠ Entire page /reset-password deopted into client-side rendering. https://nextjs.org/docs/messages/deopted-into-client-rendering /reset-password

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\6772.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}
TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info]    Generating static pages (6/26)


[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info] TypeError: fetch failed
    at node:internal/deps/undici/undici:12618:11
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
  cause: AggregateError [ECONNREFUSED]: 
      at internalConnectMultiple (node:net:1116:18)
      at afterConnectMultiple (node:net:1683:7)
      at TCPConnectWrap.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'ECONNREFUSED',
    [errors]: [ [Error], [Error] ]
  }
}

[info] 
Error occurred prerendering page "/". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at W (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\chunks\6772.js:1:7098)
    at we (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:262655)
    at C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:273539
    at Array.toJSON (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:276836)
    at stringify (<anonymous>)
    at Oe (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265286)
    at Me (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:265817)
    at Timeout._onTimeout (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:263635)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

[info]    Generating static pages (12/26)


[info]    Generating static pages (19/26)


[info] 
Error occurred prerendering page "/products". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of null (reading 'map')
    at g (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\server\app\(pages)\products\page.js:1:3577)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:139346)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:141594)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Jn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:157188)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155665)
    at Kn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:158656)
    at Wn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:151722)
    at Gn (C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\node_modules\next\dist\compiled\next-server\app-page.runtime.prod.js:1:155403)

[info]  ✓ Generating static pages (26/26)

[info] 

[info] > Export encountered errors on following paths:
	/(pages)/products/page: /products
	/page: /

[debug] [2024-09-15T03:22:05.831Z] Error: ENOENT: no such file or directory, open 'C:\Users\wou12\OneDrive\Desktop\work\react_course\eCommerce\.next\export-marker.json'
[error] 
[error] Error: An unexpected error has occurred.
